llpc/lower/llpcSpirvLowerRayTracing.cpp

```diff
@@ -572,7 +577,7 @@ bool SpirvLowerRayTracing::runImpl(Module &module) {
         replaceGlobal(global, m_globalCallableData);
         continue;
       } else if (global->getName().startswith(RtName::ShaderRecordBuffer)) {
-        processShaderRecordBuffer(global, insertPos);
+        processShaderRecordBuffer(global, m_dispatchRaysInfoDesc, m_shaderRecordIndex, insertPos);
```

Comment:
getDispatchRaysInfoDesc should be always called at the start the entry function. and not in the "if" expression. if the  ShaderRecordBuffer is not executed.

Reply:
Hmm yes, and maybe my initial suggestion was not a good one. Still, having a method `getDispatchRaysInfoDesc()` implies that it could just be called from anywhere and work correctly, but as-is this isn't true because it uses a builder call and depends on the correct insert point.

How about just removing `getDispatchRaysInfoDesc()` entirely and:

* directly insert the CreateLoadBufferDesc call in the appropriate place above
* access `m_dispatchRaysInfoDesc` directly everywhere